internal/cmd/controller/agentmanagement/controllers/cluster/import.go

```diff
@@ -77,47 +87,60 @@ func RegisterImport(
 	clusters.OnChange(ctx, "import-cluster", h.OnChange)
 	fleetcontrollers.RegisterClusterStatusHandler(ctx, clusters, "Imported", "import-cluster", h.importCluster)
 	config.OnChange(ctx, h.onConfig)
+
+	clustersCache := clusters.Cache()
```

Comment:
I wonder how expensive the `Cache()` operation might be; would it make sense to do this?
```suggestion
	clustersCache := h.clustersCache
```

Reply:
Good question.
It's actually not expensive, since using `.Cache()` on the same `Controller` will provide the same underlying storage:
https://github.com/rancher/wrangler/blob/main/pkg/generic/controller.go#L271-L277
`Cache` is just a wrapper around the indexer from the sharedInformer configured by the controller.

Having said that, this is only "cheap" because we already had handlers/controllers registered for the `Cluster` kind. If we only had a Client, converting it into a Cache will indeed incur into the costs of populating that cache.

Reply:
Thanks for clarifying :bow: .